scripts/image_compression.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+#
+# Copyright 2025 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to compress images in the repository using GraphicsMagick."""
+
+from __future__ import annotations
+
+import logging
+import pathlib
+import subprocess
+import tempfile
+
+from PIL import Image # pylint: disable=import-error
+from typing import List, TypedDict, Union
+
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+def check_compressable_images(
+        input_path: Union[str, pathlib.Path]
+    ) -> List[CompressedImageInfo]:
+    """Check and compress images using GraphicsMagick."""
+
+    result_images: List[CompressedImageInfo] = []
+    supported_extensions = {'.png', '.jpg', '.jpeg', '.webp'}
+
+    for file_path in pathlib.Path(input_path).glob('**/*.*'):
+        if file_path.suffix.lower() not in supported_extensions:
+            continue
+
+        try:
+            with Image.open(file_path):
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    temp_compressed = (
+                        pathlib.Path(tmpdir) / f'compressed_{file_path.name}'
+                    )
+
+                    if file_path.suffix.lower() == '.png':
+                        cmd = [
+                            'gm', 'convert',
+                            str(file_path),
+                            '-strip',
+                            '-compress', 'Zip',
+                            str(temp_compressed)
+                        ]
+                    elif file_path.suffix.lower() in {'.jpg', '.webp'}:
```

Comment:
Where is .jpeg handled?

Probably better if you define IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION and IMAGE_EXTENSIONS_SUPPORTING_LZW_COMPRESSION as top-level constants, and have ALL_IMAGE_EXTENSIONS be an addition of those two lists, and refer to those instead.

Also try to minimize branching. Change the arg to -compress based on the type and define the cmd only a single time.